LoopKit/DosingDecisionStore.swift

```diff
@@ -84,12 +86,29 @@ public class DosingDecisionStore {
         completion?(nil)
     }
     
-    public func destroy() {
-        self.store.managedObjectContext.performAndWait {
-            let coordinator = self.store.managedObjectContext.persistentStoreCoordinator!
-            let store = coordinator.persistentStores.first!
-            let url = coordinator.url(for: store)
-            try! coordinator.destroyPersistentStore(at: url, ofType: NSSQLiteStoreType, options: nil)
+    private static var encoder: PropertyListEncoder = {
+        let encoder = PropertyListEncoder()
+        encoder.outputFormat = .binary
+        return encoder
+    }()
+
+    private func encodeDosingDecision(_ dosingDecision: StoredDosingDecision) -> Data? {
+        do {
+            return try Self.encoder.encode(dosingDecision)
+        } catch let error {
```

Comment:
[nit] I don't think you need the `let error` anymore in Swift.  `error` automatically is available